### PR TITLE
[fix](mtmv) Fix written by mv successfully but not get mv lock. If use mv, data maybe wrong

### PR DESCRIPTION
**Pull request by seawinde**
## Proposed changes
When refresh partition mv, meanwhile, the query that can be written successfully by the same mv,  if we run the query, the result maybe wrong.
this pr fix this.

### CODE REVIEW

**Comment by seawinde** [comment]
run buildall

**Review by 924060929 — DISMISSED** [review]

**Comment by seawinde** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/AbstractMaterializedViewRule.java [hunk 1]
```diff
@@ -352,11 +352,17 @@ protected List<Plan> doRewrite(StructInfo queryStructInfo, CascadesContext casca
...
-            recordIfRewritten(queryStructInfo.getOriginalPlan(), materializationContext);
             trySetStatistics(materializationContext, cascadesContext);
             rewriteResults.add(rewrittenPlan);
             // if rewrite successfully, try to regenerate mv scan because it maybe used again
             materializationContext.tryReGenerateScanPlan(cascadesContext);
+            //  if rewrite successfully, try to get mv read lock to avoid data inconsistent
+            if (materializationContext instanceof AsyncMaterializationContext && !materializationContext.isSuccess()) {
+                // Need get mtmv read lock, to avoid the data of plan which use mv is wrong
+                cascadesContext.getStatementContext()
+                        .addTableReadLock(((AsyncMaterializationContext) materializationContext).getMtmv());
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/AbstractMaterializedViewRule.java:363]: should lock table before rewrite

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by morrySnow** [comment]
run feut

**Review by morrySnow — COMMENTED** [review]
could add some regression test case?

**Comment by seawinde** [comment]
run buildall

**Review by zddr — APPROVED** [review]

**Review by 924060929 — DISMISSED** [review]

**Comment by seawinde** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Comment by morrySnow** [comment]
run p0

**Pull request merged**